whylog/constraints/__init__.py

```diff
@@ -0,0 +1,36 @@
+class Constraint(object):
```

Comment:
Which testing do you mean?
- testing if entering rule match with causes and effect currently processing by user?
- testing which causes and effects can be generally found after adding new rule?

I think the second one should be processed more less like normal investigation. So it is not exactly teacher's responsibility.